ssg/templates.py

```diff
@@ -207,6 +207,18 @@ def build_lang_for_templatable(self, templatable, lang):
         filled_template = self.get_lang_contents_for_templatable(templatable, lang)
         self.write_lang_contents_for_templatable(filled_template, lang, templatable)
 
+    def build_platform(self, platform):
+        """
+        Builds templated content of a given Platform (all CPEs/Symbols) for all available
+        languages, writing the output to the correct build directories.
+        """
+        for symbol in platform.test.get_symbols():
+            platform.test.pass_parameters(self.product_cpes)
+            cpe = self.product_cpes.get_cpe(symbol.as_id())
+            if cpe.is_templated():
+                for lang in self.get_resolved_langs_to_generate(cpe):
+                    self.build_lang_for_templatable(cpe, lang)
```

Comment:
Some of this code shouldn't be in the template class, because it interacts with low-level details of `Platform.test` and `CPEItem`. And you call `platform.test.pass_parameters(self.product_cpes)` the same way for each symbol, is it intended?
You probably want to "pass parameters" to the platform as a whole, and then get list of templated CPEs from it without having to know which buttons to push exactly in order to get those.

Reply:
Fixed.